hooks/tk-multi-publish2/maya/publish_lod_geometry.py

```diff
@@ -209,6 +237,83 @@ def publish_files(self, task_settings, item, publish_path):
             instances.
         :param item: Item to process
         :param publish_path: The output path to publish files to
+        """
+        # Creating a temporary file on the publish path, where the alembic from maya would be
+        # exported
+        publish_file_temp = tempfile.NamedTemporaryFile(mode='w+b',
+                                                        suffix='.abc',
+                                                        prefix='tmp',
+                                                        dir="/var/tmp/",
+                                                        delete=True)
+
+        publish_path_temp = publish_file_temp.name.replace("\\", "/")
+
+        # Deciding what type of alembic needs to be exported, based on the current plugin name
+        current_plugin = self.plugin.name
+
+        if current_plugin == "Publish Geometry":
+            # Exporting alembic to a temp location.
+            # This will later be renamed and written to the publish_path
+            self._export_abc_cache(task_settings, item, publish_path_temp)
+
+            # ...and execute it:
+            try:
+                # ensure the publish folder exists:
+                publish_folder = os.path.dirname(publish_path)
+                ensure_folder_exists(publish_folder)
+
+                # Renaming top group name to be the asset name, in exported alembic.
+                asset_name = item.context.entity["name"]
+                self._rename_abc_top_group(publish_path_temp, str(publish_path), asset_name)
+            except Exception as e:
+                raise Exception("Failed to export Geometry: %s" % e)
+
+            self.logger.debug(
+                "Exported group %s to Alembic '%s'." % (item.properties.fields["node"],
+                                                        publish_path)
+            )
+        elif current_plugin == "Publish GPU Cache":
+            current_lod_item = item.get_property("lod_full_name")
+
+            # Clearing the selection for gpu cache export and only selecting the group that needs
+            # to be exported
+            cmds.select(clear=True)
```

Comment:
Should we store the current selection and restore it after we are done?

Reply:
I thought of it, But would there be an actual need of it?
I couldn't think of a reason that why the artist would need the selection to be restored.
Let me know, if there is something specific that they may have mentioned in the past and i am unaware.